src/test/java/org/breedinginsight/api/v1/controller/TraitUploadControllerIntegrationTest.java

```diff
@@ -78,11 +81,55 @@ public void setup() {
         // Insert user into program as not active
         dsl.execute(fp.get("InsertInactiveProgramUser"));
 
+        // Insert Trait
+        dsl.execute(fp.get("InsertProgramObservationLevel"));
+        dsl.execute(fp.get("InsertProgramOntology"));
+        dsl.execute(fp.get("InsertMethod"));
+        dsl.execute(fp.get("InsertScale"));
+        dsl.execute(fp.get("InsertTrait"));
+
         // Retrieve our new data
         validProgram = programDao.findAll().get(0);
 
     }
 
+    @Test
```

Comment:
I think we should also add a test case for the blank row case prior to data and row numbers in errors. It looks like there is an issue in the excel parser for this case that will need to be fixed.

[empty_then_2_rows.xlsx](https://github.com/Breeding-Insight/bi-api/files/5008188/empty_then_2_rows.xlsx)

Reply:
Added a test for this. I also fixed it in the parsing service. This will now pass and that empty row will be ignored.